examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 2: Memory clean-up\n",
+      "\n",
+      "\n",
+      "In the next example we show the ability of the associative memory to retrieve a noisy pattern. \n",
+      "This is a very common use of autoassociative memories, also called \"clean-up\" as the AM removes the noise in the input.\n",
+      "\n",
+      "Now, instead of presenting just one semantic pointer from a vocabulary, we present a combination of several semantic pointers. This is still one single vector, which is mostly similar to one semantic pointer but is also somewhat similar to other semantic pointers. Therefore, as the input is not \"clean\" we can say that we are presenting a noisy pattern.\n",
+      "The effect would have been similar if we had chosen randomly distributed numbers using, for example, `numpy`'s `random.rand` function. To indicate that the output of the AM should be similar to just one stored vector, we set the parameter `wta_output` to `True`. `WTA` is a computational principle called winner-take-all, stating that one, mostly active element should be regarded as the winner among possible, less similar alternatives."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "with nengo.Network('Cleanup', seed=1) as model2:\n",
+      "    assoc_mem = spa.AssociativeMemory(input_vocab=vocab, wta_output=True)\n",
+      "    \n",
+      "    input_vector = 0.2*vocab['APPLE'].v + 0.7*vocab['CHERRY'].v + 0.5*vocab['APRICOT'].v\n",
+      "    input_node = nengo.Node(output=input_vector, label='input')\n",
+      "    nengo.Connection(input_node, assoc_mem.input)\n",
+      "\n",
+      "    input_probe = nengo.Probe(input_node)\n",
+      "    output_probe = nengo.Probe(assoc_mem.output, synapse=0.03)\n",
+      "\n",
+      "sim = nengo.Simulator(model2)\n",
+      "sim.run(0.2)\n",
+      "t = sim.trange()\n",
+      "\n",
+      "input_net = sim.data[input_probe]\n",
+      "output_net = sim.data[output_probe]\n",
+      "\n",
+      "plot_similarities(input_net, output_net, vocab)\n"
+     ],
+     "language": "python",
+     "metadata": {
+      "scrolled": true
+     },
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We can see that the input presented to the network is mostly similar to the semantic pointer `CHERRY`, but it is also similar to other semantic pointers in the vocabulary (e.g. `APRICOT` and `APPLE`). At the output, the network successfully represents a pattern similar to `CHERRY`. If we wanted to get a pattern at the output whose similarity with the most similar vocabulary item is one, we could have used the `threshold_output` parameter when creating the AM module and setting it to `True`."
```

Comment:
Maybe instead of "As we have specified the threshold of 0.4 the resulting output vector will be similar to APPLE and CHERRY as their similarity with the input vector is greater than the threshold." write "Since we have specified the threshold of 0.4, and APPLE and CHERRY are both above this threshold, both semantic pointers will be output at their original similarity values."